packages/terra-date-picker/src/DatePicker.jsx

```diff
@@ -159,6 +184,44 @@ class DatePicker extends React.Component {
     }
   }
 
+  handleOnInputFocus(event) {
+    if (this.props.onInputFocus) {
+      this.props.onInputFocus(event);
+    }
+
+    if (!this.isDefaultDateAcceptable) {
+      this.handleChange(null, event);
+      this.isDefaultDateAcceptable = true;
+    }
+  }
+
+  handleOnCalendarButtonClick(event, onClick) {
+    if (this.props.onCalendarButtonClick) {
+      this.props.onCalendarButtonClick(event);
+    }
+
+    if (!this.isDefaultDateAcceptable && !this.validateDefaultDate()) {
+      this.handleChange(null, event);
+    } else if (onClick) {
+      onClick();
```

Comment:
perhaps we should pass the event to onClick?

Reply:
This onClick is actually react-datepicker's [onInputClick](https://github.com/Hacker0x01/react-datepicker/blob/master/src/datepicker.jsx#L474) and it doesn't take any [parameters](https://github.com/Hacker0x01/react-datepicker/blob/master/src/datepicker.jsx#L326-L330).

Reply:
^ Can we add inline documentation indicate this onClick doesn't take parameters?

Reply:
Added inline comment: https://github.com/cerner/terra-core/pull/974/commits/619c1d70775ed668582319dfa0410181bc8e4e55

Reply:
Thanks @benbcai